Assets/Scripts/SS3D/Systems/Chat/ChatMessageSender.cs

```diff
@@ -0,0 +1,146 @@
+﻿using FishNet;
+using FishNet.Connection;
+using JetBrains.Annotations;
+using SS3D.Core;
+using SS3D.Logging;
+using SS3D.Permissions;
+using SS3D.Systems.Entities;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace SS3D.Engine.Chat
+{
+    public static class ChatMessageSender
```

Comment:
Make this a NetworkSystem or a System. Avoid static classes managing network stuff, will also solve the need to have a InitializeIfNeeded because you'll already be initializing when the system does.